File: 1501~2000/1963/Minimum Number of Swaps to Make the String Balanced.swift
class Solution {
    func minSwaps(_ s: String) -> Int {
        var n : Int = s.count, r : Int = n - 1, cnt : Int = 0, ans : Int = 0
        var arr = Array(s)
        for i in 0..<n{
            if i > r{
                break
            }
            else if arr[i] == "["{
                cnt += 1
            }
            else if cnt > 0{
                cnt -= 1
            }
            else{
                while arr[r] == "]"{
                    r -= 1
                }
                cnt += 1
                ans += 1
            }
        }
        return ans
    }
}

File: 1501~2000/1963/Minimum Number of Swaps to Make the String Balanced.cpp
class Solution {
public:
    int minSwaps(string s) {
        int n = s.size(), r = n - 1, cnt = 0, ans = 0;
        for(int i = 0;i < n;i ++){
            if(s[i] == '[')
                cnt ++;
            else{
                if(cnt > 0)
                    cnt --;
                else{
                    while(s[r] == ']')
                        r --;
                    s[r] = ']';
                    cnt ++;
                    ans ++;
                }
            }
        }
        return ans;
    }
};
